File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            remainder = target - nums[i]
            print(remainder)
            for j in range(i+1,len(nums)):
                print(i, j, nums[j])
                if(remainder == nums[j]):
                    return([i,j])
        
        return []

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int low = 0;
        int n = nums.size();
        int high = n-1;
        while(low<high){
            int mid = low+(high-low)/2;
            if(nums[mid]<nums[high]){
                high = mid;
            }else{
                low = mid+1;
            }
        }

        int pivot = low;
        if(target>=nums[pivot] && target<=nums[n-1]){
            low = pivot;
            high = n-1;
        }else{
            low = 0;
            high = pivot-1;
        }
        while(low<=high){
            int mid = low+(high-low)/2;
            if(target == nums[mid]) return mid;
            if(target<nums[mid]){
                high=mid-1;
            }else{
                low = mid+1;
            }
        }
        return -1;
    }
};
